pkg/adaptor/rethinkdb/reader.go

```diff
@@ -0,0 +1,197 @@
+package rethinkdb
+
+import (
+	"errors"
+	"sync"
+
+	"github.com/compose/transporter/pkg/client"
+	"github.com/compose/transporter/pkg/log"
+	"github.com/compose/transporter/pkg/message"
+	"github.com/compose/transporter/pkg/message/ops"
+
+	re "gopkg.in/gorethink/gorethink.v3"
+)
+
+var (
+	_ client.Reader = &Reader{}
+)
+
+// Reader fulfills the client.Reader interface for use with both copying and tailing a RethinkDB
+// database.
```

Comment:
this Reader can handle straight copies as well as fire up changefeeds to be used for tailing.